src/config/middleware.js

```diff
@@ -10,14 +10,41 @@ const checkSessionAndRedirect = (req, res, next) => {
 
   // If the user is on the Home Page and does not have a session, set it
   if (isOnHomepage && !hasSession) {
+    console.log(
+      `user has no session but is on the HP , so lets set the session id`
+    );
     req.session.userSession = {
       startedJourney: true,
     };
   }
 
-  // If the user doesn't have a session and is not on the homepage, redirect to Journey Guard Page
+  // If the user doesn't have a session and is not on the homepage, redirect to Journey Guard Page but if entering on / take them to welcome page
   if (!hasSession && !isOnHomepage) {
-    return res.render("journeyGuard.njk");
+    if (req.path === "/") {
+      console.log(
+        `user is not on HP and does not have a session, the path is ${req.path} so lets redirect to welcome`
+      );
+      return res.redirect("./welcome");
+    } else {
+      console.log(
+        `user is not on HP and does not have a session, the path is ${req.path}`
+      );
+      return res.render("journeyGuard.njk");
+    }
+  }
+  // If the user has a session and the path is / redirect to Home Page path which is "/welcome"
+  if (hasSession && req.path === "/") {
+    console.log(
+      `user has session but the path is ${req.path}, let redirect to /welcome`
+    );
+    return res.redirect("/welcome");
+  }
+
+  if (!hasSession && req.path === "/") {
```

Comment:
This "if" is not necessary because if no session and req.path = / , you will valid the "if" line 22 (no session && not homepage).
So I don't think the code will never run this one.